pkg/autodiscovery/listeners/docker.go

```diff
@@ -79,7 +80,7 @@ func NewDockerListener() (ServiceListener, error) {
 		filters:    filters,
 		services:   make(map[string]Service),
 		stop:       make(chan bool),
-		health:     health.RegisterReadiness("ad-dockerlistener"),
+		health:     health.RegisterLiveness("ad-dockerlistener"),
```

Comment:
Is it expected to move back to Liveness probe?

Reply:
Yes it is.
In #7579, I moved some components from a liveness probe to a readiness probe because those components could become unhealthy because of something external to the agent (a network failure or docker daemon becoming unresponsive for ex.). And we don’t want the liveness probe to become unhealthy for a reason external to the agent.

But thanks to the introduction of the context deadline in this PR, the `ad-dockerlistener` component cannot become unhealthy because of an external component anymore.
If the docker daemon is unresponsive, the context deadline will expire, the docker API call will be aborted, the error will be logged and the `ad-dockerlistener` will properly respond to the health checker.
So, now, the `ad-dockerlistener` can become unhealthy only in case of bug like a deadlock or a premature goroutine exit.
That’s why it’s back to a liveness probe.